datafusion/sql/src/expr/mod.rs

```diff
@@ -603,6 +604,25 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
         if !fields.is_empty() {
             return not_impl_err!("Struct fields are not supported yet");
         }
+
+        if values
+            .iter()
+            .any(|value| matches!(value, SQLExpr::Named { .. }))
+        {
+            self.create_named_struct(values, input_schema, planner_context)
```

Comment:
I think the values in `create_named_struct` will be `SQLExpr::Named` in this PR, so maybe we can also make a change here?
https://github.com/apache/arrow-datafusion/blob/d8d521ac8b90002fa0ba1f91456051a9775ae193/datafusion/sql/src/expr/mod.rs#L610-L631

Reply:
Oh, ignore this. One case is only some values are Named so no change is needed.